src/lib.rs

```diff
@@ -1,6 +1,8 @@
 use serde::{Deserialize, Serialize};
 use std::io::{stdout, Write};
 use std::mem::size_of;
+use std::sync::Arc;
+use tokio::sync::Mutex;
```

Comment:
Why do you need tokio Mutex instead of the standard one?

Reply:
to my understanding it's "better"(?) since e.g. the ssh channel arc<mutex<>> can be awaited by two separate threads